src/Tasks/ResGenDependencies.cs

```diff
@@ -90,11 +89,25 @@ internal bool UseSourcePath
             }
         }
 
+        public ResGenDependencies() { }
+
+        public ResGenDependencies(ITranslator translator)
+        {
+            Translate(translator);
+        }
+
+        public override void Translate(ITranslator translator)
+        {
+            resXFiles.Translate(translator, typeof(ResXFile));
+            portableLibraries.Translate(translator, typeof(PortableLibraryFile));
+            translator.Translate(ref baseLinkedFileDirectory);
+            translator.Translate(ref _serializedVersion);
+        }
+
         internal ResXFile GetResXFileInfo(string resxFile, bool useMSBuildResXReader)
         {
             // First, try to retrieve the resx information from our hashtable.
-            var retVal = (ResXFile)resXFiles.GetDependencyFile(resxFile);
-            if (retVal == null)
+            if (resXFiles.GetDependencyFile(resxFile) is not ResXFile retVal || retVal == null)
```

Comment:
I don't feel like this is clearer than switching the cast to an `as`.

Reply:
That's probably better, thanks. This wasn't about clarity but correctness—the cast was failing and throwing an exception because it wasn't a ResXFile. That shouldn't happen in practice, and it's now fixed (see previous comment), but did cause problems for a bit.